executor/builder.go

```diff
@@ -4664,3 +4667,30 @@ func (b *executorBuilder) validCanReadTemporaryTable(tbl *model.TableInfo) error
 
 	return nil
 }
+
+func (b *executorBuilder) readFromCache(tblInfo *model.TableInfo, startTS uint64) kv.MemBuffer {
+
+	tbl, ok := b.is.TableByID(tblInfo.ID)
+	if !ok {
+		b.err = errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(b.ctx.GetSessionVars().CurrentDB, tblInfo.Name))
+		return nil
+	}
+	cacheData := tbl.(table.CachedTable).TryReadFromCache(startTS)
+	if cacheData != nil {
+		return cacheData
+	} else {
+		go func() {
+			defer func() {
+				if r := recover(); r != nil {
```

Comment:
can you add some log ？

Reply:
okay

Reply:
You can refer here.

https://github.com/pingcap/tidb/blob/7f36a07de9682b37d46240b16a2107f5c84941ba/util/misc.go#L88

Reply:
okay